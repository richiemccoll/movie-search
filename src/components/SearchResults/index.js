import React, { Fragment } from "react";
import Card from "../Card";

function SearchResults({ results }) {
  return (
    <Fragment>
      {results.length > 0 ? (
        <ul>
          {results.map((result, idx) => (
            <li key={`${result.title}-${idx}`}>
              <Card
                image={result.image_url}
                title={result.title}
                overview={result.overview}
                releaseDate={result.release_date}
                popularity={result.popularity}
              />
            </li>
          ))}
        </ul>
      ) : null}
      <style jsx>
        {`
          ul {
            display: grid;
            grid-template-columns: repeat(12, 1fr);
            max-width: 1120px;
            padding: 1em;
            margin: 0 auto;
          }

          li {
            grid-column: span 12;
            margin-bottom: 1em;
          }

          @media (min-width: 700px) {
            li {
              grid-column: span 6;
            }
          }

          @media (min-width: 1020px) {
            li {
              grid-column: span 4;
            }
          }
        `}
      </style>
    </Fragment>
  );
}

export default SearchResults;
