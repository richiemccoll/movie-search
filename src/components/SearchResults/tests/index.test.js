import React from "react";
import renderer from "react-test-renderer";

import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import SearchResults from "..";

Enzyme.configure({ adapter: new Adapter() });

describe("SearchResults", () => {
  it("should render null if no results", () => {
    const result = renderer.create(<SearchResults results={[]} />);
    expect(result).toMatchSnapshot();
  });

  it("should render a list of results", () => {
    const results = [1, 2, 3].map(i => ({
      image_url: `image-url-${i}`,
      title: `title-${i}`,
      overview: `overview-${i}`,
      release_date: `release-date-${i}`,
      popularity: i,
    }));

    const result = renderer.create(<SearchResults results={results} />);
    expect(result).toMatchSnapshot();
  });
});
