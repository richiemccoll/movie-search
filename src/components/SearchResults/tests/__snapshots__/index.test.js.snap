// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults should render a list of results 1`] = `
<ul
  className="jsx-455658503"
>
  <li
    className="jsx-455658503"
  >
    <article
      className="jsx-3065145673"
    >
      <img
        alt=""
        className="jsx-3065145673"
        src="image-url-1"
      />
      <h2
        className="jsx-3065145673"
      >
        title-1
      </h2>
      <p
        className="jsx-3065145673"
      >
        overview-1
      </p>
      <small
        className="jsx-3065145673"
      >
        Release Date - 
        release-date-1
      </small>
      <small
        className="jsx-3065145673"
      >
        Popularity - 
        1
         / 10 
      </small>
    </article>
  </li>
  <li
    className="jsx-455658503"
  >
    <article
      className="jsx-3065145673"
    >
      <img
        alt=""
        className="jsx-3065145673"
        src="image-url-2"
      />
      <h2
        className="jsx-3065145673"
      >
        title-2
      </h2>
      <p
        className="jsx-3065145673"
      >
        overview-2
      </p>
      <small
        className="jsx-3065145673"
      >
        Release Date - 
        release-date-2
      </small>
      <small
        className="jsx-3065145673"
      >
        Popularity - 
        2
         / 10 
      </small>
    </article>
  </li>
  <li
    className="jsx-455658503"
  >
    <article
      className="jsx-3065145673"
    >
      <img
        alt=""
        className="jsx-3065145673"
        src="image-url-3"
      />
      <h2
        className="jsx-3065145673"
      >
        title-3
      </h2>
      <p
        className="jsx-3065145673"
      >
        overview-3
      </p>
      <small
        className="jsx-3065145673"
      >
        Release Date - 
        release-date-3
      </small>
      <small
        className="jsx-3065145673"
      >
        Popularity - 
        3
         / 10 
      </small>
    </article>
  </li>
</ul>
`;

exports[`SearchResults should render null if no results 1`] = `null`;
